File: 12.LessonTwelve-Dictionary/src/models/Student.java
package models;

public class Student {
	public String id;
	public String name;
	public String surname;

	public Student(String id, String name, String surname) {
		this.id = id;
		this.name = name;
		this.surname = surname;
	}

	@Override
	public String toString() {
		return "Id: " + id + ", Name: " + name + ", Surname: " + surname;
	}
}

File: 36.LessonThirtySix-JavaScript/student.js
class Student {
    constructor(name, surname) {
        this.name = name;
        this.surname = surname;
        this.exams = [];
        this.average = 0;
    }

    addExam(exam) {
        this.exams.push(exam);
        this.updateAverage();
    }

    deleteExam(examIndex) {
        this.exams.splice(examIndex, 1);
        this.updateAverage();
    }

    updateAverage(){
        let grade = 0;
        this.exams.forEach((exam) => grade += exam.grade);
        this.average = grade / this.exams.length;
    }

    toString() {
        return `Student: ${this.name} ${this.surname}, with an average grade of: ${this.average}`;
    }

}